OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs

```diff
@@ -121,10 +123,26 @@ public override Activity Tick(Actor self)
 				}
 
 				if (!onRailsHack)
-					attack.Target = Target.Invalid;
+					attack.Key.Target = Target.Invalid;
 
 				return NextActivity;
 			}
+
+			IEnumerable<KeyValuePair<AttackFollow, Armament>> SelectActiveAttack(Target t, bool forceAttack)
+			{
+				foreach (var a in attackArray)
+				{
+					if (!a.IsTraitDisabled)
+					{
+						foreach (var w in a.ChooseArmamentsForTarget(t, forceAttack))
+						{
+							yield return new KeyValuePair<AttackFollow, Armament>(a, w);
+						}
+					}
+				}
```

Comment:
Pretty sure all of those curly brackets aren't supposed to be here, correct me if I'm wrong, bit fuzzy on inner loops with an if in the convention.